src/generateDocumentation.js

```diff
@@ -0,0 +1,311 @@
+'use strict';
+
+const YAML = require('yaml');
+const fileUtils = require('./fileUtils.js');
+
+module.exports = {
+  generateDocumentation: function () {
+    const template = (this.customVars && this.customVars.documentation) ? this.customVars.documentation : {};
+    const version = getOpenApiVersion(this.options.exportType);
+
+    const documentation = {};
+    if (version === "2.0") {
+      documentation.swagger = version;
+    } else {
+      documentation.openapi = version;
+    }
+    documentation.info = generateInfo(template.api);
+    documentation.paths = generatePaths(version, this.serverless.service);
+
+    const models = generateModels(template.models);
+    if (version === "2.0") {
+      if (models && Object.keys(models).length > 0) {
+        documentation.definitions = models;
+      }
+      if (template.securitySchemes) {
+        documentation.securityDefinitions = template.securitySchemes;
+      }
+    } else {
+      if ((models && Object.keys(models).length > 0) || template.securitySchemes) {
+        documentation.components = {
+          schemas: models,
+          securitySchemes: template.securitySchemes || {},
+        };
+      }
+    }
+
+    if (template.api && template.api.tags) {
+      documentation.tags = template.api.tags;
+    }
+
+    const extension = fileUtils.getFileExtension(this.options.outputFileName);
+    let output;
+    if (fileUtils.isYamlFileExtension(extension)) {
+      output = YAML.stringify(documentation, { indent: 2 });
+    } else {
+      output = JSON.stringify(documentation, null, 2);
+    }
+    this.fs.writeFileSync(this.options.outputFileName, output);
```

Comment:
Is there any reason why is this synchronous? It might be ok in the context of a serverless plugin but technically the event loop started turning by now...

Reply:
Mmm.. I though that Serverless commands (and their lifecycle events) were executed synchronously, which is why I did it this way. Anyway, changed to write the output file asynchronously: https://github.com/failsafe-engineering/serverless-aws-apigateway-documentation/pull/43/commits/51a3c0a456c7c6c938f413139ef367aad49cbb60

Reply:
I'm wondering why I can't see the change in the PR. Perhaps you have to close the PR and reopen again?

Reply:
Mmm.. that's strange. I can see it: https://github.com/failsafe-engineering/serverless-aws-apigateway-documentation/pull/43/commits/51a3c0a456c7c6c938f413139ef367aad49cbb60

Conversation:
![Screenshot 2024-01-11 120411](https://github.com/failsafe-engineering/serverless-aws-apigateway-documentation/assets/3950191/b15bf2f3-21fc-4bb7-9d96-4c78034d69d6)

Commits
![Screenshot 2024-01-11 120123](https://github.com/failsafe-engineering/serverless-aws-apigateway-documentation/assets/3950191/0ddc2670-d5fb-4084-8df4-e7d69b9b4e8a)